bin/prefix-vendor-namespace.php

```diff
@@ -45,92 +96,165 @@
 // Read our composer.json file into an array.
 $composer_json = json_decode( file_get_contents( dirname( __DIR__ ) . '/composer.json' ), true );
 
-// Flag modified files that maybe should have been modified more.
-$file_notices = [];
+foreach ( $packages as $package ) {
+	// Process namespaces and uses in all package files.
+	foreach ( find_files( $package['package'] ) as $file ) {
+		process_file( $file, $package, true, true );
+	}
 
-foreach ( $replacements as $namespace => $path ) {
-	$files = find_files( $path );
+	// Process only uses in dependent package files.
+	foreach ( find_dependent_files( $package['package'] ) as $file ) {
+		process_file( $file, $package, false, true );
+	}
 
-	$quoted = preg_quote( $namespace, '#' );
-	foreach ( $files as $file ) {
-		$contents = file_get_contents( $file );
+	// Prefix autoloads and remove file autoloads from vendor/composer/installed.json
+	update_autoloads(
+		"{$vendor_dir}/composer/installed.json",
+		$composer_json['autoload']['files'] ?? [],
+		$package
+	);
+}
 
-		// Check to see whether a replacement has already run for this namespace. Just in case.
-		if ( false !== strpos( $contents, "{$new_namespace}\\{$namespace}" ) ) {
-			continue 2;
-		}
 
-		$namespace_change = 0;
-		$uses_change      = 0;
+/**
+ * Process prefixing in a specific file.
+ *
+ * @since x.x.x
+ *
+ * @param string $file             File name.
+ * @param string $package          Package prefix configuration.
+ * @param bool   $prefix_namespace Whether to prefix namespaces or not.
+ * @param bool   $prefix_uses      Whether to prefix uses of a namespace.
+ */
+function process_file( $file, $package, $prefix_namespace = true, $prefix_uses = false ) {
```

Comment:
seems for me that $prefix_namespace should be an array. But its type is defined as Srtring.

Reply:
`$prefix_namespace` should still be a bool, but you are right that the second parameter `$package` got refactored from `string > array`

Reply:
Yes. sorry. I messes up the names 😂